pkg/pool-aclamm/contracts/AclAmmPool.sol

```diff
@@ -180,6 +189,47 @@ contract AclAmmPool is BalancerPoolToken, PoolInfo, Version, IBasePool, BaseHook
         return _lastTimestamp;
     }
 
+    function setNewQ0(uint256 newQ0, uint256 startTime, uint256 endTime) external {
+        _initSqrtQ0 = _updateAndGetSqrtQ0();
+        _targetQ0 = newQ0;
+        _startChangingQ0Timestamp = startTime;
+        _endChangingQ0Timestamp = endTime;
+    }
+
+    function _updateAndGetSqrtQ0() internal returns (uint256) {
+        uint256 storedSqrtQ0 = _sqrtQ0;
+        uint256 currentSqrtQ0 = _calculateCurrentSqrtQ0(storedSqrtQ0);
+        if (currentSqrtQ0 != storedSqrtQ0) {
+            _setSqrtQ0(currentSqrtQ0);
+        }
+
+        return currentSqrtQ0;
+    }
```

Comment:
We don't need to update the _sqrtQ0 on every interaction. We can just calculate the current Q0 based on start and end values and times.

We should only change the storage when a pool manager wants to start updating the Q0 to another value, so the `Q0State` (Similar to StablePool's AmplificationState) needs to be updated.